Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -438,16 +470,72 @@ String CSSCounterStyleDescriptors::systemCSSText() const
     case System::EthiopicNumeric:
     case System::DisclosureClosed:
     case System::DisclosureOpen:
-        return emptyString();
+        break;
     }
-    return emptyString();
+    return false;
+}
+
+String CSSCounterStyleDescriptors::systemCSSText() const
+{
+    if (!m_explicitlySetDescriptors.contains(ExplicitlySetDescriptors::System))
+        return emptyString();
+    return serialize(m_system, m_isExtendedResolved, m_extendsName, m_fixedSystemFirstSymbolValue, SerializeUsingMakeString { });
+}
+
+void CSSCounterStyleDescriptors::systemCSSText(StringBuilder& builder) const
+{
+    if (!m_explicitlySetDescriptors.contains(ExplicitlySetDescriptors::System))
+        return;
+    return serialize(m_system, m_isExtendedResolved, m_extendsName, m_fixedSystemFirstSymbolValue, SerializeUsingStringBuilder { builder });
+}
+
+// MARK: - Negative
+
+bool CSSCounterStyleDescriptors::NegativeSymbols::hasCSSText() const
+{
+    return true;
+}
+
+String CSSCounterStyleDescriptors::NegativeSymbols::cssText() const
```

Comment:
It kind of seems like if the implementation is just using a `StringBuilder`, then it shouldn’t be a member function. Caller can use a helper to make the string in an idiom that is not too wordy, but having the function makes it seem like it’s more efficient than it really is? Maybe.